sentry_sdk/integrations/redis.py

```diff
@@ -7,13 +7,64 @@
 from sentry_sdk._types import MYPY
 
 if MYPY:
-    from typing import Any
+    from typing import Any, Sequence
 
 _SINGLE_KEY_COMMANDS = frozenset(
     ["decr", "decrby", "get", "incr", "incrby", "pttl", "set", "setex", "setnx", "ttl"]
 )
 _MULTI_KEY_COMMANDS = frozenset(["del", "touch", "unlink"])
 
+#: Trim argument lists to this many values
+_MAX_NUM_ARGS = 10
+
+
+def patch_redis_pipeline(pipeline_cls, is_cluster, get_command_args_fn):
+    # type: (Any, bool, Any) -> None
+    old_execute = pipeline_cls.execute
+
+    def sentry_patched_execute(self, *args, **kwargs):
+        # type: (Any, *Any, **Any) -> Any
+        hub = Hub.current
+
+        if hub.get_integration(RedisIntegration) is None:
+            return old_execute(self, *args, **kwargs)
+
+        with hub.start_span(op="redis", description="redis.pipeline.execute") as span:
+            with capture_internal_exceptions():
+                span.set_tag("redis.is_cluster", is_cluster)
+                transaction = self.transaction if not is_cluster else False
+                span.set_tag("redis.transaction", transaction)
+
+                commands = []
+                for i, arg in enumerate(self.command_stack):
+                    if i > _MAX_NUM_ARGS:
+                        break
+                    command_args = []
+                    for j, command_arg in enumerate(get_command_args_fn(arg)):
+                        if j > 0:
+                            command_arg = repr(command_arg)
+                        command_args.append(command_arg)
+                    commands.append(" ".join(command_args))
```

Comment:
SDK folks: What prevents us from using list comprehensions and/or slicing for this kind of code? Do we support Python < 2.0?

Reply:
I suspect this was my code: It's a performance thing. Don't want to allocate unnecessarily large intermediate lists